code/WorkInProgress/AzrunStuff.dm

```diff
@@ -239,3 +239,353 @@
+				owner.current.delStatus("bound_ghost")
+				if(old_brain)
+					owner.brain = old_brain // attempt to restore to previous brain?
+				owner = null
+
+	on_life()
+		var/mob/living/M = holder.donor
+		if(!ishuman(M)) // silicon shouldn't have these problems
+			return
+
+		if(M.client && (isnull(M.client.color) || M.client.color == "#FFFFFF") && !ON_COOLDOWN(src,"ghost_eyes", 5 MINUTES))
+			boutput(M,"Your vision starts to change as your connection this body wavers.")
+			animate(M.client, color=COLOR_MATRIX_GRAYSCALE, time=5 SECONDS, easing=SINE_EASING)
+			animate(color=COLOR_MATRIX_IDENTITY, time=30 SECONDS, easing=SINE_EASING)
+		if(prob(1))
+			boutput(M,"You find you lose control of your body for a moment...")
+			M.changeStatus("paralysis", 2 SECONDS)
+		if(prob(1))
+			boutput(M,"You suddenly feel sluggish as though your connection to your body isn't as strong.")
+			M.changeStatus("slowed", 8 SECONDS, 2)
+
+	get_movement_controller()
+		.= MC
+
+	Crossed(atom/movable/AM)
+		..()
+
+		var/mob/dead/observer/O = AM
+		if(!src.owner && !GET_COOLDOWN(src,"ghost_suck") && istype(O))
+			if(jobban_isbanned(O, "Ghostbrain"))
+				boutput(O, "<span class='notice'>Sorry, you are banned from playing a ghostbrain.</span>")
+				return
+			if(O.can_respawn_as_ghost_critter()) // Azrun TODO New func with more apporpriate verbage?
+				actions.start(new/datum/action/bar/capture_ghost(O), src)
+				ON_COOLDOWN(src, "ghost_suck", 2 SECONDS)
+
+	Entered(atom/movable/A,atom/OldLoc)
+		. = ..()
+		var/mob/dead/observer/O = A
+		if(MC && istype(O) )
+			O.use_movement_controller = src
+			O.setStatus("bound_ghost", duration = 2 MINUTES, optional=list("anchor"=src, "client"=O.client))
+			if (istype(O.abilityHolder, /datum/abilityHolder/ghost_observer))
+				var/datum/abilityHolder/ghost_observer/GH = O.abilityHolder
+				GH.disable(TRUE)
+				GH.updateButtons()
+
+	Exited(atom/movable/A,atom/OldLoc)
+		. = ..()
+		var/mob/dead/observer/O = A
+		if(istype(O) && O.use_movement_controller == src)
+			O.use_movement_controller = null
+
+		if (istype(O) && istype(O.abilityHolder, /datum/abilityHolder/ghost_observer))
+			var/datum/abilityHolder/ghost_observer/GH = O.abilityHolder
+			GH.disable(FALSE)
+			GH.updateButtons()
+
+/datum/action/bar/capture_ghost
+	id = "capture_ghost"
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_ACTION
+	duration = 5 SECONDS
+	var/mob/dead/observer/target
+	var/image/pulling
+
+	New(mob/dead/observer/O)
+		..()
+		if (istype(O))
+			target = O
+		if(!pulling)
+			pulling = image('icons/effects/effects.dmi',"pulling",pixel_y=16)
+			pulling.alpha = 200
+
+	onUpdate()
+		..()
+		if(GET_DIST(target,owner) != 0)
+			interrupt(INTERRUPT_ALWAYS)
+
+	onStart()
+		..()
+		boutput(target, "<span class='notice'>You feel yourself being pulled into [owner]!</span>")
+		owner.UpdateOverlays(pulling, id)
+
+	onEnd()
+		..()
+		var/obj/item/organ/brain/ghost/B = owner
```

Comment:
Shouldn't here somewhere be some check for whether there isn't another ghost already controlling the brain? Or is having multiple ghosts in a brain intended? Or am I missing something?

Reply:
Check existed earlier in callstack.  Adding checks so nothing slips by.